lib/devices.go

```diff
@@ -0,0 +1,238 @@
+package usb_tree_lib
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/google/gousb"
+	"github.com/google/gousb/usbid"
+)
+
+const (
+	StateNormal  LogState = "normal"
+	StateAdded   LogState = "added"
+	StateRemoved LogState = "removed"
+)
```

Comment:
Generally I'd say the order should be

- imports
- types
- consts
- vars
- funcs